core/common/src/thrift/block_master.thrift

```diff
@@ -3,6 +3,26 @@ namespace java alluxio.thrift
 include "common.thrift"
 include "exception.thrift"
 
+enum BlockMasterInfoField {
+  LIVE_WORKER_NUM
+  LOST_WORKER_NUM
+  CAPACITY_BYTES
+  USED_BYTES
+  FREE_BYTES
+  CAPACITY_BYTES_ON_TIERS
+  USED_BYTES_ON_TIERS
+}
+
+struct BlockMasterInfo {
+ 1: i32 liveWorkerNum
```

Comment:
@aaudiber would it be useful to make these optional?

Reply:
Optional keyword seems to be more useful when we have the default value for a variable but our variables do not have default values.

By using the set<BlockMasterInfoField> filter, we make all the variables optional.

Reply:
My experience in the past was to follow a policy of using optional 100% of the time. It worked well and we didn't have to worry about differences between default requiredness and optional requiredness. The current code base is using a mix of default and optional, so I'd say just be consistent within the file and we'll standardize on something in 2.0. Just make sure not to set any fields as required

Reply:
In our code base,  many TOptions and TResponse in file_system_master.thrift and file_system_worker.thrift use optional, but all the Info class (like previous MasterInfo, WorkerInfo, Command, LocalityTier, WorkerNetAddress, LineageInfo) use default.
I preferred to keep default at that time.
Thanks